30 lint entire development

JimtheBurger opened the pull request:
# Proposed Changes

* Updates all code to follow the lint
* Adds Linting to README

IMPORTANT TO NOTE: GitHub will still validate if the lint detects a Warning, but not a Problem. (I think it's fine, but if we want to be that specific we can look into it)

You can check `Actions` on the branch to see the Lint working in GitHub.

Fixes #30 .

JimtheBurger committed:
trying some lint stuff

JimtheBurger committed:
Updated README for linting

JimtheBurger commented on adversarial_apps/src/components/navbar.tsx:
```diff
@@ -1,42 +1,6 @@
-import { usePathname } from 'next/navigation';
-import Link from 'next/link';
 
-const links = [
```
ESLint did not like these since it was unused. It was a placeholder.

JimtheBurger committed:
removed unused comment

SavvyDolphin77 commented on adversarial_apps/README.md:
```diff
@@ -34,3 +34,11 @@ You can check out [the Next.js GitHub repository](https://github.com/vercel/next
+
+## Linting
+
+It's recommended to install an ESLint extension for these to be handled during development.
```
Need to change this to say and be more accurate: ESLint comes provided as part of the required packages for application. You must use that version of ESLint. When you install the packages as seen above in README you will have ESLint and can run the following command:

SavvyDolphin77 replied:
Double check the lint to have it pass all linting checks with recent update to action

JimtheBurger replied:
They Validate now. I excluded `node_module`s from the black check since we are `.gitignoring` those files anyway.

SavvyDolphin77 committed:
Updates Action To Check Not Fix

Caught that it was actually running the linter rather then checking them, fixed that issue.

JimtheBurger committed:
changed readme and added npm run lint

JimtheBurger committed:
Merge branch '30-lint-entire-development' of https://github.com/AdverseApps/Adversarial-Apps-Web into 30-lint-entire-development

JimtheBurger committed:
changed action to exlcude node_modules

SavvyDolphin77 committed:
Moved README To Github README

Changes where in the wrong README, should have been in the one in the root project. Carried that over and also added commands for black and isort as  well as explaining the other option for linting with ESLint
